### ISSUE TITLE
How to incorporate output from parse_data_tdt?

### ISSUE THREAD

**Issue by JRJacoby** [csn-le/wave_clus#57]
Hey all, thanks so much for making this tool. I'm working with a TDT system and the powerpoint tutorial said that I'd probably have to use the scripts in the tools folder in order to process the data first. I got that working and got the tdt_meta_data output, but what do I do with that output? Additionally, which TDT files do I load into wave_clus? It won't let me load a whole tank or block because it sees them as regular folders. Thanks for any help!

EDIT: I found all the individual channel files that are also part of the output, doy. But it seems you need to feed parse_data_tdt the raw waveform as opposed to spikes? I've already completed my spike detection and would like to be able to use those detected spikes for the clustering. Is there a way to use parse_data_tdt with spike information as opposed to wave information? I've tried loading a .mat file that I made with "spikes" and "index" variables directly into wave_clus but got an error saying: "Subscript indicies must either be real positive integers or logicals."

**Comment by ferchaure** [comment]
But it seems you need to feed parse_data_tdt the raw waveform as opposed to spikes?
Yes, We didn't add support for spikes in a TDT system.

Could you tell me where is that error? Did you check that the length of the spikes are par.w_pre+par.w_post ?
Are the spikes aligned?  in that case you have to set par.interpolation = 'n';

**Comment by JRJacoby** [comment]
Ah, it was the size of the waveform not matching the parameters! Thank you so much! One last question for my own understanding - your tutorial and readme talk about how to use the temperature setting to adjust clusters, but temperature isn't something I've run into before. What exactly is this value a measure of? Thanks again!

**Comment by ferchaure** [comment]
The temperature is related to the scale that the SPC uses for find the clusters. 
At high temperature, SPC finds the clusters that have low density and a big gap between them. At low temperature it finds the clusters dense and close together.

**Comment by JRJacoby** [comment]
Thanks! I'm still having difficulty on a couple of points here. Point me
towards documentation if I'm missing something but I've found the
powerpoint tutorial and the README so far. I tried loading in one of the
.tdtch files and got the following:

Error using ellip (line 60)
The cutoff frequencies must be within the interval of (0,1).

Error in spike_detection_filter (line 12)
    [b,a] = ellip(par.detect_order,0.1,40,[fmin_detect fmax_detect]*2/sr);

Error in readInData/get_segment (line 234)
                xf_detect = spike_detection_filter(x(1:lplot), obj.par);

Error in wave_clus>load_data_button_Callback (line 211)
            x = data_handler.get_segment();

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in wave_clus (line 63)
    gui_mainfcn(gui_State, varargin{:});

Error while evaluating UIControl Callback.

I went and edited the detect_fmax  and the sort_fmax so that it would come
out to below one (although this required lowing it all the way to 500 which
seems low to me) and then ran it again and got the following error:

Subscripted assignment dimension mismatch.

Error in amp_detect (line 81)
                index(nspk) = iaux + xaux(i) -1;

Error in wave_clus>load_data_button_Callback (line 212)
            [new_spikes, temp_aux_th, new_index]  = amp_detect(x,
            handles.par);

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in wave_clus (line 63)
    gui_mainfcn(gui_State, varargin{:});

Error while evaluating UIControl Callback.


I tried instead creating my own .mat file with "data" and "sr", which got
me:

Error using horzcat
Dimensions of matrices being concatenated are not consistent.

Error in amp_detect (line 106)
xf = [xf zeros(1,w_post)];

Error in wave_clus>load_data_button_Callback (line 212)
            [new_spikes, temp_aux_th, new_index]  = amp_detect(x,
            handles.par);

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in wave_clus (line 63)
    gui_mainfcn(gui_State, varargin{:});

Error while evaluating UIControl Callback.

I'd be happy using either the .tdtch files or my own .mat files but I'd
love to find a way for this to work. It would be nice having all my
analysis in MATLAB. Thanks again for taking the time.

Best,
John
